src/main/java/net/minecraftforge/fml/common/registry/EntityEntry.java

```diff
@@ -44,14 +45,7 @@ public EntityEntry(Class<? extends Entity> cls, String name)
     //Protected method, to make this optional, in case people subclass this to have a better factory.
     protected void init()
     {
-        try
-        {
-            this.ctr = this.cls.getConstructor(World.class);
-        }
-        catch (NoSuchMethodException e)
-        {
-            throw new RuntimeException("Invalid class " + this.cls + " no constructor taking " + World.class.getName());
-        }
+        this.ctr = ReflectionHelper.findConstructor(this.cls, World.class);
```

Comment:
You should provide a way to change this via the builder (by providing a `Function<World, E>`.